main.tf

```diff
@@ -0,0 +1,88 @@
+provider "alicloud" {
+  profile                 = var.profile != "" ? var.profile : null
+  shared_credentials_file = var.shared_credentials_file != "" ? var.shared_credentials_file : null
+  region                  = var.region != "" ? var.region : null
+  skip_region_validation  = var.skip_region_validation
+  configuration_source    = "terraform-alicloud-modules/emr-hadoop"
+}
+
+data "alicloud_emr_instance_types" "default" {
+  destination_resource  = "InstanceType"
+  cluster_type          = "HADOOP"
+  support_local_storage = false
+  instance_charge_type  = var.charge_type
+  support_node_type     = ["MASTER", "CORE", "TASK", "GATEWAY"]
+}
+
+data "alicloud_emr_disk_types" "data_disk" {
+  destination_resource = "DataDisk"
+  cluster_type         = "HADOOP"
+  instance_charge_type = var.charge_type
+  instance_type        = data.alicloud_emr_instance_types.default.types.0.id
+  zone_id              = data.alicloud_emr_instance_types.default.types.0.zone_id
+}
+
+data "alicloud_emr_disk_types" "system_disk" {
+  destination_resource = "SystemDisk"
+  cluster_type         = "HADOOP"
+  instance_charge_type = var.charge_type
+  instance_type        = data.alicloud_emr_instance_types.default.types.0.id
+  zone_id              = data.alicloud_emr_instance_types.default.types.0.zone_id
+}
+
+resource "random_uuid" "this" {}
+
+resource "alicloud_ram_role" "default" {
+  name     = "terraform-ram-role-${random_uuid.this.result}"
```

Comment:
`name     = "substr("tf-ram-role-for-hadoop-${replace(random_uuid.this.result, "-", "")}", 0, 32)"